src/IonParserTextRaw.ts

```diff
@@ -757,41 +756,56 @@ export class ParserTextRaw {
   }
 
   private _read_string3() : void {
-    var ch;
-    if (this._read() != CH_SQ || this._read() != CH_SQ) {
-      this._error( "expected triple quote" );
-    }
-    this._start = this._in.position();
-    for(;;) {  // read sequence of triple quoted strings
-      for(;;) { // read single quoted strings until we see the triple quoted terminator
-        this._read_string_helper(CH_SQ, true);
-        // if it's not a triple quote, it's just content
-        if (this._read() == CH_SQ && this._read() == CH_SQ) {
-          this._end = this._in.position() - 3;
-          break;
-        }
-      }
-      ch = this._read_after_whitespace(true);
-      if (ch != CH_SQ) {
-        this._unread(ch);
-        break;
-      }
-      if (this._peek("\'\'") == ERROR) {
-        break;
+      let ch : number;
+      this._unread(this._peek(""));
+      // read sequence of triple quoted strings:
+        //not exiting correctly, need to put a correcting boolean to check for the end. just finds error probably due to the read calls or something.
+      for(this._start = this._in.position() + 3; this._peek("\'\'\'") !== ERROR; this._in.unread(this._read_after_whitespace(true))) {//need to check if this off by ones on print?
+          this._read();
+          this._read();
+          this._read();
+          //in tripleQuotes
+          //index content of current triple quoted string,
+          //looking for more triple quotes
+          while(this._peek("\'\'\'") === ERROR) {
+              ch = this._read();
+              if( ch === EOF) throw new Error('Closing triple quotes not found.');
+            // read single quoted strings until we see the triple quoted terminator
+            // if it's not a triple quote, it's just content
+          }
+          //mark the possible end of the series of triplequotes
+
+          // Set the end of the value,
+          // this._end will reset later if further triple quotes are found after indexing through whitespace,
+          this._end = this._in.position();
+          //Index past the triple quote.
+
+          this._read();
```

Comment:
second instance of 3 identical lines for reading 3 elements. Make a helper that takes the number of `next` 's to call and use that.